### DISCUSSION TITLE
Is there any documentation for OCPP-2.0.1

### DISCUSSION THREAD

**Discussion by odiwml147** [ChargeTimeEU/Java-OCA-OCPP#314] [Q&A]
I saw that the library supports version OCPP-2.0.1 on the homepage,do you have any documents related to OCPP-2.0.1?
########################################################################################
There are only 1.6 usage documents,I want to know how to use this project with OCPP-2.0.1.

**Comment by TVolden** [comment]
Hi @odiwml147,

There is an official specification here:
https://www.openchargealliance.org/protocols/ocpp-201/
The implementation is mostly structured after the official specification.

What are you looking for, in particular?

**Reply by TVolden** [reply]
I has come to my attention that OCA changed their homepage and thereby killed the link, here's the new one:
https://openchargealliance.org/protocols/open-charge-point-protocol/#OCPP2.0.1

**Comment by odiwml147** [comment]
> Hi @odiwml147,
> 
> There is an official specification here: https://www.openchargealliance.org/protocols/ocpp-201/ The implementation is mostly structured after the official specification.
> 
> What are you looking for, in particular?



Hello @TVolden .
I have a general understanding of some 2.0 specifications.There are only 1.6 usage documents,I want to know how to use this project with OCPP-2.0.1.

**Comment by bdevia** [comment]
Hi,

I am also interested in knowing how to use OCPP-2.0.1. I was doing a little research and in the thread: https://github.com/ChargeTimeEU/Java-OCA-OCPP/pull/239, this capability of the library is mentioned, however, it is not mentioned how to use it :c

**Comment by jmluy** [comment]
hi @bdevia and @odiwml147. This might help you start with implementing a 2.0.1 server https://github.com/ChargeTimeEU/Java-OCA-OCPP/blob/master/ocpp-v2-test/src/main/java/eu/chargetime/ocpp/test/OCPP201MultiProtocolFakeCSMS.java. 
@robert-s-ubi might have more information

**Comment by TVolden** [comment]
I have moved this to the new discussion section.

**Comment by robert-s-ubi** [comment]
In addition to the source @jmluy mentioned, there is also a fake OCPP 1.6/2.0.1 Charging Station here:

https://github.com/ChargeTimeEU/Java-OCA-OCPP/blob/master/ocpp-v2-test/src/main/java/eu/chargetime/ocpp/test/OCPP201MultiProtocolFakeChargingStation.java

And an integration test which creates both a fake OCPP 1.6/2.0.1 CSMS and Charging Station and sends a boot notification:

https://github.com/ChargeTimeEU/Java-OCA-OCPP/blob/master/ocpp-v2-test/src/test/java/eu/chargetime/ocpp/test/OCPP201MultiProtocolIntegrationTest.java

Beyond that, the library works as it did before OCPP 2.0.1 support was added.

**Comment by bdevia** [comment]
Hi @TVolden @jmluy @robert-s-ubi , thank you very much for your answers!,

Right now I only have one last question, how can I access the OCPP 2.0.1 version packages? Checking the maven repository, they only have the packages:
- Java OCA OCPP V1.6
- Java OCA OCPP Common
- Java OCA OCPP OCPP J

None of them include the OCPP 2.0.1 version, so I would like to know if they will include this version in the maven repository, or failing that, I would have to manually download the project from Git to be able to use it.

**Reply by TVolden** [reply]
I haven't had time to publish it to maven yet. Sorry.

That said you can use jit pack:
https://jitpack.io/#ChargeTimeEU/Java-OCA-OCPP/v1.2

**Reply by jmluy** [reply]
There is a discussion to start doing a 2.0 release https://github.com/ChargeTimeEU/Java-OCA-OCPP/issues/364

**Comment by odiwml147** [comment]
Thanks!
I cloned the code and made some local maven images. Now, I am referring to the testing module for development.



At 2024-02-26 16:28:52, "Thomas Volden" ***@***.***> wrote:

I haven't had time to publish it to maven yet. Sorry.

That said you can use jit pack:
https://jitpack.io/#ChargeTimeEU/Java-OCA-OCPP/v1.2
